robot/impl/resource_manager.go

```diff
@@ -830,30 +838,57 @@ func (manager *resourceManager) updateResources(
 	defer manager.configLock.Unlock()
 	var allErrs error
 
+	// stores resource names so that if there is duplicate we error
+	seenResources := make(map[string]bool)
+
 	for _, s := range conf.Added.Services {
 		rName := s.ResourceName()
 		if manager.opts.untrustedEnv && rName.API == shell.API {
 			allErrs = multierr.Combine(allErrs, errShellServiceDisabled)
 			continue
 		}
+		if err := manager.validateUniqueResource(seenResources, rName.String()); err != nil {
```

Comment:
actually only want to do this with new modules as if we check for modified, they will always exist in the resource graph. 

Did a test to check that if we changed the name of a module while viam-server is running so there is a duplicate, it will show an error in the logs.

Reply:
> if we changed the name of a module

Yeah modified resources are only "modified" if they retain the same name. Otherwise they're just added IIRC.